File: crypto/src/main/java/io/oversec/one/crypto/symbase/BaseSymmetricEncryptionParams.java
package io.oversec.one.crypto.symbase;

import io.oversec.one.crypto.AbstractEncryptionParams;
import io.oversec.one.crypto.EncryptionMethod;

import java.util.ArrayList;
import java.util.List;

public abstract class BaseSymmetricEncryptionParams extends AbstractEncryptionParams {
    protected List<Long> mKeyIds = new ArrayList<>();

    protected BaseSymmetricEncryptionParams(EncryptionMethod method, String coderId, String padderId) {
        super(method, coderId, padderId);
    }


    public List<Long> getKeyIds() {
        return mKeyIds;
    }


}

File: crypto/src/main/java/io/oversec/one/crypto/symbase/BaseSymmetricEncryptionParams.kt
package io.oversec.one.crypto.symbase

import io.oversec.one.crypto.AbstractEncryptionParams
import io.oversec.one.crypto.EncryptionMethod
import java.util.ArrayList

abstract class BaseSymmetricEncryptionParams protected constructor(
    method: EncryptionMethod,
    coderId: String,
    padderId: String?
) : AbstractEncryptionParams(method, coderId, padderId) {
    var keyIds: List<Long> = ArrayList()
        protected set


}
